### ISSUE TITLE
Add a TTS engine

### ISSUE THREAD

**Issue by yacineMTB** [yacineMTB/talk#3]
This thing needs to respond back to us on some event.
Right now, the strategy to reduce latency is to generate precanned responses constantly. Maybe we can also follow the same strategy with some TTS system?

Ideally this would
- be abstracted in some function, takes some text and responds with audio

For now we can just save it as a wav file. The scope of this task is figuring out what reasonable candidates we have for TTS, with one of the goals being low latency.

**Comment by yacineMTB** [comment]
I am considering
- https://github.com/MycroftAI/mimic3
- https://openreview.net/forum?id=__czv_gqDQt
- https://github.com/heatz123/naturalspeech
- https://github.com/legekka/GanyuTTS
- https://github.com/rendchevi/nix-tts

**Comment by KKcorps** [comment]
Should consider this as well

https://github.com/suno-ai/bark

**Comment by abacaj** [comment]
What do you think of https://github.com/rhasspy/piper - was pretty straightforward to set up. I haven't been able to train it but the voices were ok

**Comment by yacineMTB** [comment]
Thank you for sharing!! I think that this is a facade that uses mimic3 under the hood. It's cpp so I should be able to churn out a binding pretty quickly for this

**Comment by yacineMTB** [comment]
Yeah i looked at the code & some samples, and asked a friend of mine

it is perfect for this

thanks @synesthesiam!!

**Comment by iacore** [comment]
I've heard that the Mycroft model is not really good. Maybe it's better to use Microsoft's TTS.

**Comment by ariym** [comment]
https://github.com/coqui-ai/TTS

**Comment by yacineMTB** [comment]
>coqui-ai

From a quick glance it seems too bloated

>Microsoft's TTS

Is it locally runnable?

>coqui-ai

From a quick glance this seems too bloated

![image](https://github.com/yacineMTB/talk/assets/10282244/74d25a72-fbdd-44a1-8595-f52060bcb5bc)

I think this is why mimicv3 wins. This is actually ridiculous. Plus i think the project is kinda based
Also, the models are highly variable based on data quality. Picking mimic, but I'll abstract the TTS portion so it's swappable.

**Comment by iacore** [comment]
https://github.com/iacore/nix-tts is pretty good

Microsoft TTS is usable on Windows machines. On Linux there is espeak, although the quality is not good.

**Comment by synesthesiam** [comment]
@yacineMTB you're welcome! I wrote both Piper and Mimic 3: Piper is the better choice as it's newer and faster :+1:

**Comment by jmanhype** [comment]
help this is not working on windows
